src/function/index.tsx

```diff
@@ -0,0 +1,78 @@
+import {FC} from 'react';
+import {v4 as uuidv4} from 'uuid';
+
+export interface ICell {
+  Amount: number;
+  ID: string;
+}
+
+export const randomGenerator = (): ICell => ({
+  Amount: Math.floor(Math.random() * (999 - 100 + 1)) + 100,
+  ID: uuidv4(),
+});
+
+export const createMatrix = (settings: {columns: number; rows: number}): Array<Array<ICell>> => {
+  // if (settings.columns && settings.rows) {
+  const arr = new Array(settings.columns * settings.rows).fill(0).map(randomGenerator);
+  console.log(arr);
+
+  let matrixRows: ICell[][] = [];
+  let start = 0;
+  for (let i = 0; i < settings.rows; i++) {
+    matrixRows = [...matrixRows, arr.slice(start, settings.columns * (i + 1))];
+    start = settings.columns * (i + 1);
+  }
+  return matrixRows;
+};
+
+export const matrixSort = (matrix: Array<Array<ICell>>): Array<ICell> =>
+  matrix
+    .flat()
+    .map((item) => item)
+    .sort((a, b) => a.Amount - b.Amount);
+
+export const sumRowNumbers = (row: Array<ICell>): number => {
+  return row.reduce((acc, el) => acc + Number(el.Amount), 0);
+};
+
+export const sumRowAvg = (avg: Array<number>): number => {
+  return avg.reduce((acc, el) => acc + Number(el), 0);
+};
+
+export const calcAvgNumbers = (columns: number, rows: number, matrixRows: Array<Array<ICell>>): Array<number> => {
+  let columnAvg: Array<number> = [];
+  for (let i = 0; i < columns; i++) {
+    let acc = 0;
+    for (let j = 0; j < rows; j++) {
+      acc += matrixRows[j][i].Amount;
+    }
+    columnAvg = [...columnAvg, Math.round(acc / Number(rows))];
+  }
+  return columnAvg;
+};
+
+export const findNearestCells = (cells: number, sortedMatrix: Array<ICell>, item: ICell): Array<ICell> => {
+  const copySortedMatrix = [...[], ...sortedMatrix];
```

Comment:
if you need to create a copy of the array, there is no need to write [...[], ...sortedMatrix], just [...sortedMatrix]